storm-core/src/clj/org/apache/storm/daemon/worker.clj

```diff
@@ -566,9 +566,12 @@
         worker-topology-context (worker-context worker)
         hooks (.get_worker_hooks topology)]
     (dofor [hook hooks]
-      (let [hook-bytes (Utils/toByteArray hook)
-            deser-hook (Utils/javaDeserialize hook-bytes BaseWorkerHook)]
-        (.start deser-hook topo-conf worker-topology-context)))))
+      (let [savedposition (.position hook)
```

Comment:
I guess we could do simpler: `(.duplicate hook)` to get duplicated ByteBuffer object, and read from that instance. It will not increase position from `hook` object.

Reply:
For sure this is a better solution.
Do I push a new commit implementing this or is there a better way to take into account your suggestion ?

Reply:
You can just push a new commit and squash the commits later

Reply:
I'm OK if you just make change to new squashed commit, since the diff will be one-liner or couple of lines. Sure I'm also OK to see new commit and squash later.